hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/helpers/DatanodeIdYaml.java

```diff
@@ -105,6 +112,8 @@ public static DatanodeDetails readDatanodeIdFile(File path)
     private String ipAddress;
     private String hostName;
     private String certSerialId;
+    private String persistedOpState;
+    private long persistedOpStateExpiryEpochSec = 0;
```

Comment:
Since zero is the default value for the long, would it make sense to enforce time; that is for each set command we need an explicit time. In that case, we can define a constant called INFINITY and set it to something like -1; but in the source, it will be very explicit when someone reads code, and in the command handler we can enforce the requirement (or perhaps better on the SCM side). So commands like decommission will have an infinite value explicitly set by SCM and it would be more obvious and explicit.

Reply:
This might be a good idea. The current "infinity" value in SCM is zero, as it checks for greater than zero expiry right now, but it could be changed to a constant easily enough.

It would be good for the command to always get a time so its consistent. I need to go back and refresh my memory on the command etc here.

Reply:
I think this is a good suggestion so I have logged it in HDDS-3304 and we can consider it when the decommission work is started up again.